security/pkg/pki/util/keycertbundle.go

```diff
@@ -171,11 +173,17 @@ func (b *KeyCertBundleImpl) CertOptions() (*CertOptions, error) {
 	if len(ids) != 1 {
 		return nil, fmt.Errorf("expect single id from the cert, found %v", ids)
 	}
+	size, err := GetRSAKeySize(b.privKey)
+	if err != nil {
```

Comment:
Should this be a non-recoverable error, does size = 0 generate a meaningful key cert pair?

Reply:
Yes, I thought we could support setting Key size outside of the function, when the privKey is not an RSA type key. Currently whenever CertOptions() is called, the key is an RSA key for sure. But is it possible the key is not RSA type in the future?

Reply:
Got it. But in case when we start to use other key type, would KeyCertBundle stil have enough info to handle both, call GetRSAKeySize()/GetKerberoseKeySize() by inspecting the key/cert bytes?

If returned CertOption's key size can either be set or unset, the caller better be aware of the need to set it (e.g, having a parameter overrideKeySize int)

Reply:
Getting size for other key types could be supported in the future. I think we can make the error returned here, to avoid unnecessary complexity. We can add support for other key types in the future.